### DISCUSSION TITLE
Gh-pages 404 errors with deploy a react app

### DISCUSSION THREAD

**Discussion by XAHTC** [community/community#23646] [Other Feature Feedback, Questions, & Ideas]
<p>I have 404 errors when I tried to deploy my react app to gh-pages. Here is my github - <a href="https://github.com/XAHTC/test-app/" rel="noopener nofollow ugc">https://github.com/XAHTC/test-app/</a>, and gh-pages - <a href="https://github.com/XAHTC/test-app/tree/gh-pages" rel="noopener nofollow ugc">https://github.com/XAHTC/test-app/tree/gh-pages</a>.</p>
<p>It is a screenshot of errors:</p>
<p><img src="https://user-images.githubusercontent.com/34131934/181078625-e987e526-450f-4a5a-b607-eac4d4b332f8.png" alt="" data-base62-sha1="vYJ7pM2Z5OAgQJSkyYXHAm8ZzpI" width="654" height="155"></p>

**Comment by FormerlyChucks** [comment]
<p>This is because their is no GitHub pages for your account. You will need to create a repository named <code>XAHTC.github.io</code> and place your code in their. You can see my site <a>here</a>.</p>
<ul>
<li><a href="https://pages.github.com/" rel="noopener nofollow ugc">Relevant link about GitHub pages</a></li>
</ul>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/xahtc">@XAHTC</a>, welcome to the GitHub Support Community!</p>
<p>I’m not seeing those errors in your Pages site anymore, however it looks like they were caused by your asset URLs missing the <code>test-app</code> repository name section.</p>
<p>When you’re deploying a React app to GitHub Pages there’s some steps you need to follow to make sure that everything works correctly—specifically adding the <code>homepage</code> variable to your <code>package.json</code> file. This means that all your URLs include the required repo name.</p>
<p>For exact steps I’d recommend reading over the documentation here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://create-react-app.dev/img/favicon/favicon.ico" class="site-icon" width="64" height="64">
      <a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank" rel="noopener">create-react-app.dev</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/2439407/181078627-51e0b7aa-c3a4-484b-9ef5-ba4c61a65be3.png" class="thumbnail" width="690" height="362" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e9d52951b6799822784160951f379815293e6f9a_2_10x10.png"></div>

<h3><a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank" rel="noopener">Deployment | Create React App</a></h3>

<p>npm run build creates a build directory with a production build of your app. Set up your favorite HTTP server so that a visitor to your site is served index.html, and requests to static paths like /static/js/main..js are served with the contents of...</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
